src/models/data.py

```diff
@@ -349,11 +349,25 @@ def __next__(self) -> Tuple[Union[np.ndarray, torch.Tensor],
 
                     cur_max_idx = min(cur_max_idx + 1, self.max_idx)
 
-                out_x.append(arrays.x)
+                out_x.append(arrays.x.historical)
+
+                if self.experiment == 'nowcast':
```

Comment:
So this is really silly because we're basically undoing the work we did in the `ds_folder_to_np` and in the `NowcastEngineer` but it does work. Really keen to hear your thoughts @gabrieltseng